server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -757,7 +757,9 @@ public void sendResponse(RestResponse response) {
         private void close() {
             // attempt to close once atomically
             if (closed.compareAndSet(false, true) == false) {
-                throw new IllegalStateException("Channel is already closed");
+                final IllegalStateException e = new IllegalStateException("Channel is already closed");
+                assert false : e; // this is always a bug, we tried to send two responses to the same channel
```

Comment:
... could we `assert allowDoubleResponses()` or something here so we can avoid having to catch this assertion in the test?

Reply:
Hmm then we have weird code here instead of the test? Doesn't matter much to me, but doesn't necessarily make things simpler. It's always annoying to test an assumed impossible code path I guess. If you feel strongly about it, happy to go for that :)

Reply:
Yeah this test is kinda bogus today, but still having a method to switch this behaviour off in tests isn't as weird as `catch AssertionError` IMO. Let's not risk accidentally catching the wrong `AssertionError` :grin: